### PR TITLE
Replace IRC with Slack in docs

### PR DESCRIPTION
**Pull request by goltermann**
Just a find and replace.

### CODE REVIEW

**Comment by yuvipanda** [comment]
Are you guys really moving to Slack off IRC? :(

**Comment by goltermann** [comment]
@k8s-bot please test this

**Comment by goltermann** [comment]
@yuvipanda yes, looks like it.  we tried it out, and most people liked the grouping of channels under one team, newer UI, and some of the notifications you can set up.

**Comment by yuvipanda** [comment]
I see. Does this mean Slack is the new official place to get support for k8s?

**Comment by eparis** [comment]
I'll throw in a vote against, even though it may not/doesn't matter.

**Comment by mvdan** [comment]
Another vote against. If connecting to IRC is too hard or the webchat is not enough, you can suggest things like irc cloud.

**Comment by bgrant0607** [comment]
Voting against doesn't help unless you explain why.

The project is too big to have a single channel -- there's too much noise and cross-talk. I, for one, could rarely afford to pay attention it. We needed new channels for GKE vs. K8s, users vs. devs, SIGs, etc. 

We either could create lots of new IRC channels, or choose a new medium. A number of people suggested Slack. For instance, the scalability SIG decided to use it. We tried it and liked it. 

Among other issues, without threading, IRC is very hard to follow when there's more than one conversation. There's also no obvious way to search the logs. I don't know how to discover new IRC channels within a general topic area.

With respect to support, we'd like people to try stackoverflow/stackexchange first. Real-time support doesn't scale, and we also get a lot of FAQs. And, regardless of the communication medium, we need the community to help more with support.

File: docs/devel/writing-a-getting-started-guide.md [hunk 1]
```diff
@@ -76,7 +76,7 @@ These guidelines say *what* to do.  See the Rationale section for *why*.
 
 If you have a cluster partially working, but doing all the above steps seems like too much work,
 we still want to hear from you.  We suggest you write a blog post or a Gist, and we will link to it on our wiki page.
-Just file an issue or chat us on IRC and one of the committers will link to it from the wiki.
+Just file an issue or chat us on [Slack](https://github.com/docs/troubleshooting.md#slack) and one of the committers will link to it from the wiki.
```
Comment (bgrant0607) [inline docs/devel/writing-a-getting-started-guide.md]: Please use relative links everywhere. ../troubleshooting.md#slack in this case.

**Comment by bgrant0607** [comment]
You need to run hack/update-generated-docs.sh.

File: docs/troubleshooting.md [hunk 2]
```diff
@@ -62,9 +62,9 @@ You may also find the Stack Overflow topics relevant:
 
 Someone else from the community may have already asked a similar question or may be able to help with your problem. The Kubernetes team will also monitor [posts tagged kubernetes](http://stackoverflow.com/questions/tagged/kubernetes). If there aren't any existing questions that help, please [ask a new one](http://stackoverflow.com/questions/ask?tags=kubernetes)!
 
-## IRC
+## <a name="slack"></a>Slack
 
-The Kubernetes team hangs out on IRC at [`#google-containers`](https://botbot.me/freenode/google-containers/) on freenode.  Feel free to come and ask any and all questions there.
+The Kubernetes team hangs out on Slack in the `#kubernetes-users` channel.  Slack requires registration, but the [Kubernetes](https://kubernetes.slack.com) team is open invitation to [join](http://slack.kubernetes.io).  Feel free to come and ask any and all questions there.
```
Comment (bgrant0607) [inline docs/troubleshooting.md]: It's not obvious which of these links the user should click on to join or participate. I suggest being more explicit, like register HERE and access the channels HERE.

File: docs/user-guide/monitoring.md [hunk 3]
```diff
@@ -85,7 +85,7 @@ Here is a snapshot of the a Google Cloud Monitoring dashboard showing cluster-wi
...
+Now that you’ve learned a bit about Heapster, feel free to try it out on your own clusters! The [Heapster repository](https://github.com/GoogleCloudPlatform/heapster) is available on GitHub. It contains detailed instructions to setup Heapster and its storage backends. Heapster runs by default on most Kubernetes clusters, so you may already have it! Feedback is always welcome. Please let us know if you run into any issues. Heapster and Kubernetes developers hang out on [Slack](https://github.com/docs/troubleshooting.md#slack). You can also reach us on the [google-containers Google Groups mailing list](https://groups.google.com/forum/#!forum/google-containers).
```
Comment (bgrant0607) [inline docs/user-guide/monitoring.md]: While you're here, heapster has moved to the kubernetes org. Please udpate the link.

google-containers@ isn't really for OSS anymore.

I suggest linking to the other suggested channels/forums in the troubleshooting doc rather than listing them all here.

**Comment by mvdan** [comment]
> Voting against doesn't help unless you explain why.

* Slack requires signing up to use, which freenode doesn't
* It makes it easy for people to [massively ping tons of people](https://slack.zendesk.com/hc/en-us/articles/202009646-Making-announcements)

Good news is that they have an IRC gateway, but it still requires signing up and it's messy when people send stuff other than text e.g. images or files. Also the whole mentioning via `nick` or `@nick` doesn't work well when using an IRC client.

> The project is too big to have a single channel -- there's too much noise and cross-talk. I, for one, could rarely afford to pay attention it. We needed new channels for GKE vs. K8s, users vs. devs, SIGs, etc.

Agreed on multiple channels, which works in either IRC or Slack.

> Among other issues, without threading, IRC is very hard to follow when there's more than one conversation.


Slack doesn't have threading, at least as far as I can see.


> I don't know how to discover new IRC channels within a general topic area.



With respect to support, we'd like people to try stackoverflow/stackexchange first. Real-time support doesn't scale, and we also get a lot of FAQs. And, regardless of the communication medium, we need the community to help more with support.

**Comment by mvdan** [comment]
> Voting against doesn't help unless you explain why.

* Slack requires signing up to use, which freenode doesn't
* It makes it easy for people to [massively ping tons of people](https://slack.zendesk.com/hc/en-us/articles/202009646-Making-announcements)

Good news is that they have an IRC gateway, but it still requires signing up and it's messy when people send stuff other than text e.g. images or files. Also the whole mentioning via `nick` or `@nick` doesn't work well when using an IRC client.

> The project is too big to have a single channel -- there's too much noise and cross-talk. I, for one, could rarely afford to pay attention it. We needed new channels for GKE vs. K8s, users vs. devs, SIGs, etc.

Agreed on multiple channels, which works in either IRC or Slack.

> Among other issues, without threading, IRC is very hard to follow when there's more than one conversation.

Slack doesn't have threading, at least as far as I can see.

> There's also no obvious way to search the logs.

You use botbot.me, which makes this easy.

> I don't know how to discover new IRC channels within a general topic area.

Freenode doesn't offer this service directly but some third parties do:
* http://irchelp.org/irchelp/chanlist/
* http://irc.netsplit.de/channels/?net=freenode

Also, regarding notifications, you can have those set up by being connected all the time. You don't need to be connected directly, you can use a third party service like IRC Cloud which then can push notifications to your devices.

**Comment by goltermann** [comment]
I've enabled IRC gateway from Slack for the Kubernetes team.  If you want to try it out, https://kubernetes.slack.com/account/gateways should give you the details you need.

**Comment by bgrant0607** [comment]
And our self-service signup page is http://slack.k8s.io

**Comment by goltermann** [comment]
@k8s-bot test this again

(just barely timed out after passing all E2E tests)
